tests/go/go-kafka-serde/client.go

```diff
@@ -267,13 +253,110 @@ func (tc *TestClient) RunTest() error {
 			}
 
 			numConsumed++
-		case kafka.Error:
-			util.Die("Failure during consumption: %v", e)
+		case *kafka.Error:
+			return e
 		default:
 			log.Debugf("Ignoring unknown message: %v", e)
 		}
 	}
 
+	log.Debugf("Consumed %v messages", numConsumed)
+
+	if numConsumed != tc.count {
+		util.Die("Did not consume expected number of messages: %v != %v", numConsumed, tc.count)
+	}
+
+	return nil
+}
+
+func (tc *TestClient) produceMessages() error {
+	numDelivered := 0
+
+	// Create a context with a cancel function
```

Comment:
nit: The code is as useful as this comment. The variable names are more useful than this comment.